src/jank/clojure/core.jank

```diff
@@ -431,6 +431,92 @@
                  #{ o }#
                );"))
 
+(defn conj!
+  ([]
+   (transient []))
+  ([coll]
+   coll)
+  ([coll x]
+   (native/raw "__value = visit_object
+                (
+                  [=](auto const typed_t, auto const head) -> object_ptr
+                  {
+                    using T = typename decltype(typed_t)::value_type;
```

Comment:
The pattern used is that `foo` becomes `typed_foo`. So, in this case, the object we're visiting is `coll` and the unerased (visited) object should be `typed_coll`. Using `T` for the type name, especially when there's only one, is just one of those C++ idioms that exists.

Reply:
Similarly, if the param is called `x` in the `defn`, let's call it `x` when we visit. Or call it something else, but let's keep that naming.